tests/main/snap-confine-from-core/task.yaml

```diff
@@ -34,5 +35,45 @@ execute: |
     get_journalctl_log | MATCH "DEBUG: restarting into"
 
     echo "Ensure snap-confine from the core snap is run"
-    # do not use "strace -f" for unknown reasons that hangs
     test-snapd-tools.echo hello
+
+
+    echo "Check if snap-confine profile generation test is applicable"
+    if [ $(aa-enabled) != "Yes" ]; then
+        if [ "$SPREAD_SYSTEM" = "ubuntu-16.04-64" ]; then
+            echo "No apparmor on ubuntu-16.04-64 is impossible"
+            echo "Test broken"
+            exit 1
+        fi
+        echo "SKIP: no strict confinement"
+        exit 0
+    fi
+
+    echo "Ensure snapd generates the right apparmor profile on restart"
+    PROFILES="$(find /etc/apparmor.d/ -maxdepth 1 -name "snap.core.*.snap-confine")"
+    if [ -z "$PROFILES" ]; then
+        echo "cannot find apparmor profiles for snap-confine from core"
+        echo "test broken"
+        ls -al /etc/apparmor.d
+        exit 1
+    fi
+
+    echo "Force system-key change"
+    systemd_stop_units snapd.service snapd.socket
+    printf '{"version":1}' > /var/lib/snapd/system-key
+    rm -f /etc/apparmor.d/snap.core.*.snap-confine
+    systemctl start snapd.service snapd.socket
+    wait_for_service snapd.service
+
+    echo "Ensure this also re-generates the snap-confine from core profile"
+    for _ in $(seq 30); do
```

Comment:
Did you actually need this seq? The way we process things (AFAIK) makes us only "ready" once security has been processed. If this is not the case perhaps we should investigate that (separately from this PR)

Reply:
Indeed, I think you are right. I was probably overly defensive here. I remove this seq.